### PR TITLE
IDE 2150 new liferay workspace wizard

### PR DESCRIPTION
**Pull request by AndyWu2015**

### CODE REVIEW

**Comment by AndyWu2015** [comment]
I didn't find a size suitable Liferay icon and just copy the liferay_logo_large.png in com.liferay.ide.ui project

**Comment by AndyWu2015** [comment]
Hi @gamerson  , there are many compiling errors in gralde.core.tests and should I try to fix them and continue to add my own unit tests ?

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
hey @AndyWu2015  yeah those gradle.core.tests are old, you can simply delete them.

File: common/plugins/com.liferay.ide.core/src/com/liferay/ide/core/util/CoreUtil.java [hunk 1]
```diff
@@ -550,12 +550,20 @@ public static void prepareFolder( IFolder folder ) throws CoreException
...
     }
 
-    public static String readPropertyFileValue( File propertiesFile, String key ) throws FileNotFoundException,
-        IOException
+    public static String readPropertyFileValue( File propertiesFile, String key )
     {
-        Properties props = new Properties();
-        props.load( new FileInputStream( propertiesFile ) );
-        return props.getProperty( key );
+        try
```
Comment (gamerson) [inline common/plugins/com.liferay.ide.core/src/com/liferay/ide/core/util/CoreUtil.java]: Use try/with/resources
```
try( FileInputStream fis = new FileInputStream( propertiesFile ) )
```

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/ProjectImportUtil.java [hunk 2]
```diff
@@ -549,99 +587,64 @@ public static IStatus validateSDKPath(final String currentPath)
...
 
-        if( !org.eclipse.core.runtime.Path.EMPTY.isValidPath( currentPath ) )
+        if( retVal.isOK() )
         {
-            retVal = ProjectCore.createErrorStatus( "\"" + currentPath + "\" is not a valid path." );
-        }
-        else
-        {
-            IPath osPath = org.eclipse.core.runtime.Path.fromOSString( currentPath );
+            LiferayWorkspaceUtil.clearWorkspace( currentPath );
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/ProjectImportUtil.java]: This shouldn't be called in a method call "validateWorkspacePath".  a call to "validate" should be deleting files, etc.

Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/ProjectImportUtil.java]: Also ClearWorkspace is only called by one class.  So we don't need to add it to a Util.  In general only add a method to a static Util standalone class if that method is being called by two different classes.

File: tools/plugins/com.liferay.ide.project.ui/plugin.xml [hunk 3]
```diff
@@ -66,6 +66,25 @@
                class="org.eclipse.core.resources.IResource">
          </selection>
       </wizard>
+      <wizard
+            category="com.liferay.ide.newWizards"
+            finalPerspective="com.liferay.ide.eclipse.ui.perspective.liferay"
+            icon="icons/e16/liferay_logo_16.png"
+            id="com.liferay.ide.project.ui.workspace.NewLiferayWorkspaceWizard"
+            name="Liferay Workspace"
+            preferredPerspectives="com.liferay.ide.eclipse.ui.perspective.liferay,org.eclipse.jst.j2ee.J2EEPerspective"
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.ui/plugin.xml]: The preferred perspective should be the new Liferay Workspace, not just the normal "liferay" perspective.

File: tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/wizard/AbstractLiferayWizard.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.wizards.newresource.BasicNewProjectResourceWizard;
+import org.eclipse.ui.wizards.newresource.BasicNewResourceWizard;
+import org.eclipse.wst.web.internal.DelegateConfigurationElement;
+
+/**
+ * @author Terry Jia
+ */
+@SuppressWarnings( "restriction" )
+public class AbstractLiferayWizard<T extends Element> extends SapphireWizard implements IWorkbenchWizard, INewWizard
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/wizard/AbstractLiferayWizard.java]: You are creating an AbstractLiferayWizard but only one other class is sub-classes that one, the new LiferayWorkspace wizard.  Just merge these two classes, unless you have a 2nd class that needs to sub-class the Abstract One.

In general... don't prematurely optimize things... don't make an Abstract class if there is only one single sub-class.  Just simply make everything in one class.  Only introduce an abstract class if there are two sub-classes.

**Comment by gamerson** [comment]
@AndyWu2015 lets squash all of these commits into just one commit since we have changed the design in several places.

**Comment by gamerson** [comment]
Please resend, thanks! <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**